.github/workflows/e2e-spin-preview.yml

```diff
@@ -24,7 +24,7 @@ jobs:
           NODE_OPTIONS: --openssl-legacy-provider
 
       - name: upload artifact
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```

Comment:
Hi @vdice, 

looking at the documentation for the update, it seems we can't upload multiple artifacts with same name. Its not entirely clear if that is the case even if multiple workflows are uploading these artifacts with same name?

Reply:
Thanks @rajatjindal.  Good point, the docs aren't super clear.  However, I'm fairly certain that the restriction is per workflow and that multiple, parallel workflows can use the same artifact name.  I tested via multiple parallel runs of eg https://github.com/vdice/fermyon-actions/actions/runs/10722654363 on my fork and all appears well.

Reply:
Ok, I cloned the workflow that doesn't require creds (`e2e-spin-setup.yml`) with both using the same artifact name and it doesn't appear to present a conflict: https://github.com/vdice/fermyon-actions/actions/runs/10722750992 and https://github.com/vdice/fermyon-actions/actions/runs/10722750993.  I think we are good.